framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -541,7 +517,14 @@ public Boolean visitDeclared_Typevar(AnnotatedDeclaredType subtype, AnnotatedTyp
 
     @Override
     public Boolean visitDeclared_Union(AnnotatedDeclaredType subtype, AnnotatedUnionType supertype, VisitHistory visited) {
-        return visitUnionSupertype(subtype, supertype, visited);
+        Types types = checker.getTypeUtils();
+        for (AnnotatedDeclaredType supertypeAltern : supertype.getAlternatives()) {
+            if (TypesUtils.isErasedSubtype(types, subtype.getUnderlyingType(), supertypeAltern.getUnderlyingType())
```

Comment:
As [documented](http://types.cs.washington.edu/checker-framework/current/api/org/checkerframework/framework/type/TypeHierarchy.html#isSubtype-org.checkerframework.framework.type.AnnotatedTypeMirror-org.checkerframework.framework.type.AnnotatedTypeMirror-), the Java types of the arguments to isSubtype are not always in a subtype relationship.  

>It would seem much simpler to add that special casing into isSubtype.

It would not be correct to do so.  For example, `isSubtype(@C Cloneable & @A List<@B String, @A List<@B String>)` if the `isErasedSubtype` is moved to `isSubtype`, then `isSubtype` called recursively on `@C Cloneable` and `@A List<@B String>` would return false which would cause `isSubtype(@C Cloneable & @A List<@B String, @A List<@B String>)` to return false as well.